### ISSUE TITLE
Spacing in object passed as function argument

### ISSUE THREAD

**Issue by bardware** [eslint/eslint#8705]
**Tell us about your environment**

* **ESLint Version:** eslint@next (v4.0.0-rc.0)
* **Node Version:** v8.0.0 (Win x86_64)
* **npm Version:** 5.0.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
{
	"env" : {
		"browser" : true,
		"jquery" : true
	},
	"rules" : {
		"brace-style" : [ "error", "1tbs", {} ],
		"comma-spacing" : [ "error", { "before" : false, "after" : true } ],
		"comma-style" : [ "error", "last" ],
		"space-in-parens" : [ "error", "always" ],
		"object-curly-spacing" : [ "error", "always" ]
	}
}
```

**What did you do? Please include the actual source code causing the issue.**
I have some historically grown code. Tabs and spaces are mixed, commas in front. I called `eslint --fix` on that file.

```js
$.ajax(
{ 	url : '/ajax.cfc?method=myMethod&_cf_nodebug=true'
,	type : 'POST'
,	data : {}
,	success : function( data )
	{
	console.dir(data);
}
,	async : false
} );

```

**What did you expect to happen?**
There's a space and a tab after the curly brace in line 2. I expect one of these gone.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$.ajax(
{ 	url : '/ajax.cfc?method=myMethod&_cf_nodebug=true',
	type : 'POST',
	data : {},
	success : function( data ) {
	console.dir( data );
},
	async : false
} );
```

The `--fix`ed version does not look differently than the input.
Tab and space remained there. In the last line curly brace and bracket are beautifully aligned separated by one space. 
I expect bracket and curly brace to from line one and two to be united on one line separated by one space.
I expect the extra space after the curly brace in line two gone.

**Comment by ljharb** [comment]
Is your code block the input, or the output? Could you confirm and provide the other one as well so we can visually compare?

**Comment by bardware** [comment]
> Is your code block the input, or the output?

The input! I updated my issue to also contain the output. They differ in the position of the comma. White spaces are identical.

**Comment by ljharb** [comment]
It's possible that not all those rules are fixable. What errors are reported after the `--fix` command?

**Comment by bardware** [comment]
> What errors are reported after the --fix command?

None, actually.

**Comment by kaicataldo** [comment]
@bardware Have you tried the `indent` rule? The rule was just rewritten for ESLint v4.0 (released today!), and the autofixer now works great!

**Comment by bardware** [comment]
Adding `"indent": ["error", "tab"]` to `.eslintrc.json` indents the whole block by one tab, including the mix of space and tab in line two and the closing brackets in the last line.
```
$.ajax(
	{ 	url : '/ajax.cfc?method=myMethod&_cf_nodebug=true',
		type : 'POST',
		data : {},
		success : function( data ) {
			console.dir( data );
		},
		async : false
	} );
```

**Comment by bardware** [comment]
When I delete both white spaces, the space and the tab and call `eslint --fix` one space is introduced between the brace and `url`. I assume this is because of the `object-curly-spacing` rule. This does not seem to correct the spacing between the brace and the member. If I introduce two spaces instead of tab and space it again results in no difference of input and output.

**Comment by kaicataldo** [comment]
What about `"indent": ["error", 2]` (for indentation of 2 spaces) and the [no-mixed-spaces-and-tabs rule](http://eslint.org/docs/rules/no-mixed-spaces-and-tabs)?

**Comment by not-an-aardvark** [comment]
I don't think this is a bug -- it's just that we don't have a rule to enforce against having both a space and a tab after a brace in an object.

**Comment by kaicataldo** [comment]
Ah, I see. So currently, `no-mixed-spaces-and-tabs` only checks for leading whitespace on each line, but doesn't anything after that. In that case, yes, this is really a feature request - either making a new rule that checks this or enhancing `no-mixed-spaces-and-tabs` with an option that would check this.

**Comment by nzakas** [comment]
Unfortunately, it looks like there wasn't enough interest from the team or community to implement this change. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to be implemented after 90 days tend to never be implemented, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue). This doesn't mean the idea isn't interesting or useful, just that it's not something the team can commit to.
